client-lite/src/ipc/rest_http_listener.cpp

```diff
@@ -36,25 +34,52 @@ void RestHttpListener::Start(const std::string& listenUrl)
             ++restPort;
         }
     }
-    DO_ASSERT(tmpListener);
-    _listener = std::move(tmpListener);
+    DO_ASSERT(tmpListener.is_open());
+    _listener = std::make_unique<boost_tcp_t::acceptor>(std::move(tmpListener));
+    _BeginAccept();
 }
 
 void RestHttpListener::Stop()
 {
-    if (_listener)
+    if (_listener && _listener->is_open())
     {
-        _listener->close().wait();
+        _listener->close();
         _listener.reset();
     }
+    DoLogInfo("RestHttpListener: connections recvd: %u", _numConnections.load());
 }
 
 std::string RestHttpListener::Endpoint() const
 {
-    return _listener->uri().to_string();
+    std::stringstream ss;
+    ss << _listener->local_endpoint();
+    return ss.str();
 }
 
 uint16_t RestHttpListener::Port() const
 {
-    return _listener->uri().port();
+    return _listener->local_endpoint().port();
+}
+
+void RestHttpListener::_BeginAccept()
+{
+    auto acceptSocket = std::make_shared<boost_tcp_t::socket>(*_io);
+    _listener->async_accept(*acceptSocket, [this, acceptSocket](const boost::system::error_code& ec) mutable
+        {
+            if (ec)
+            {
+                DoLogError("Async accept failed, error = %d", ec.value());
+                // TODO(shishirb) terminate process by throwing exception?
```

Comment:
Would it make sense to try again up until n failures?

Reply:
Maybe. The only time I've seen a failure here is during shutdown notifying us that socket is closed (expected). So I'll leave it as is until we see other retry-able errors.